### PR TITLE
fix edit-box error in gameview mode

### PR DESCRIPTION
**Pull request by dogeFu**
Re: #https://github.com/cocos/3d-tasks/issues/12784

### Changelog

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.

  > Manual trigger with `@cocos-robot run test cases` afterward.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: cocos/ui/editbox/edit-box-impl.ts [hunk 1]
```diff
@@ -286,12 +273,8 @@ export class EditBoxImpl extends EditBoxImplBase {
...
-        let widthRatio = 1;
-        let heightRatio = 1;
-        if (legacyCC.GAME_VIEW) {
-            widthRatio = legacyCC.gameView.canvas.width / legacyCC.game.canvas.width;
-            heightRatio = legacyCC.gameView.canvas.height / legacyCC.game.canvas.height;
-        }
+        const widthRatio = 1;
+        const heightRatio = 1;
         scaleX *= widthRatio;
         scaleY *= heightRatio;
```
Comment (PPpro) [inline cocos/ui/editbox/edit-box-impl.ts:279]: 建议直接去掉这一段没有意义的代码

File: cocos/ui/editbox/edit-box-impl.ts [hunk 2]
```diff
@@ -182,33 +182,20 @@ export class EditBoxImpl extends EditBoxImplBase {
     }
 
     private _addDomToGameContainer () {
-        if (legacyCC.GAME_VIEW && this._edTxt) {
-            legacyCC.gameView.container.appendChild(this._edTxt);
-            legacyCC.gameView.head.appendChild(this._placeholderStyleSheet!);
-        } else if (game.container && this._edTxt) {
+        if (game.container && this._edTxt) {
```
Comment (PPpro) [inline cocos/ui/editbox/edit-box-impl.ts:185]:
```suggestion
        if (this._edTxt) {
```

File: cocos/ui/editbox/edit-box-impl.ts (hunk 1 above)
Comment (PPpro) [inline cocos/ui/editbox/edit-box-impl.ts]: 建议直接干掉这一段

Comment (PPpro) [reply cocos/ui/editbox/edit-box-impl.ts]: 底下的 `offsetX += viewport.x * widthRatio / dpr;`  改为 `offsetX += viewport.x  / dpr;`

Comment (dogeFu) [reply cocos/ui/editbox/edit-box-impl.ts]: OK

File: cocos/ui/editbox/edit-box-impl.ts (hunk 2 above)
Comment (PPpro) [reply cocos/ui/editbox/edit-box-impl.ts:185]: 抱歉这块还是需要做判断的

Comment (dogeFu) [reply cocos/ui/editbox/edit-box-impl.ts:185]: 好

**Review by PPpro — APPROVED** [review]

**Review by SantyWang — APPROVED** [review]

**Pull request merged**